nautobot_golden_config/jobs.py

```diff
@@ -272,31 +258,68 @@ class Meta:
         description = "Process to run all Golden Configuration jobs configured."
         has_sensitive_variables = False
 
-    @gc_repos
-    def run(self, *args, **data):  # pylint: disable=unused-argument
+    def run(self, *args, **data):  # pylint: disable=unused-argument, too-many-branches
         """Run all jobs on a single device."""
-        if constant.ENABLE_INTENDED:
-            config_intended(
-                self.job_result,
-                self.logger.getEffectiveLevel(),
-                self,
-                self.qs,
-                self.device_to_settings_map,
-            )
-        if constant.ENABLE_BACKUP:
-            config_backup(
-                self.job_result,
-                self.logger.getEffectiveLevel(),
-                self.qs,
-                self.device_to_settings_map,
-            )
-        if constant.ENABLE_COMPLIANCE:
-            config_compliance(
-                self.job_result,
-                self.logger.getEffectiveLevel(),
-                self.qs,
-                self.device_to_settings_map,
-            )
+        self.logger.debug("Compiling device data for GC job.", extra={"grouping": "Get Job Filter"})
+        self.qs = get_job_filter(data)
+        self.logger.debug(
+            f"In scope device count for this job: {self.qs.count()}", extra={"grouping": "Get Job Filter"}
+        )
+        self.logger.debug("Mapping device(s) to GC Settings.", extra={"grouping": "Device to Settings Map"})
+        self.device_to_settings_map = get_device_to_settings_map(queryset=self.qs)
+        gitrepo_types = ["backup_repository", "jinja_repository", "intended_repository"]
+        self.logger.debug(
+            f"Repository types to sync: {', '.join(sorted(gitrepo_types))}",
+            extra={"grouping": "GC Repo Syncs"},
+        )
+        current_repos = get_refreshed_repos(job_obj=self, repo_types=gitrepo_types, data=self.qs)
```

Comment:
Its a nit pick, and i know we've kind of went away from requiring DRY method specifically to tackle this problem.  But both `run all jobs` and the decorator have these identical lines.  Can we make them a function and call that function from all 3 spots instead.  if its going to be a pain or require a bunch of rework I'm not willing to hold up this PR for it.

Reply:
Refactored as requested. Let me know if there's anything else that needs to be addressed.